android/app/src/androidTest/java/com/github/quarck/calnotify/dismissedeventsstorage/EventDismissTest.kt

```diff
@@ -0,0 +1,253 @@
+import com.github.quarck.calnotify.logs.DevLog
+import com.github.quarck.calnotify.testutils.MockApplicationComponents
+import com.github.quarck.calnotify.testutils.MockCalendarProvider
+import com.github.quarck.calnotify.testutils.MockContextProvider
+import com.github.quarck.calnotify.testutils.MockTimeProvider
+import io.mockk.*
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.Assert.*
+import org.junit.Ignore
+
+@RunWith(AndroidJUnit4::class)
+class EventDismissTest {
+    private val LOG_TAG = "EventDismissTest"
+    
+    private lateinit var mockContext: Context
+    private lateinit var mockDb: EventsStorageInterface
+    private lateinit var mockComponents: MockApplicationComponents
+    
+    @Before
+    fun setup() {
+        DevLog.info(LOG_TAG, "Setting up EventDismissTest")
+
+        // Setup mock database
+        mockDb = mockk<EventsStorageInterface>(relaxed = true)
+        
+        // Setup mock providers
+        val mockTimeProvider = MockTimeProvider()
+        val mockContextProvider = MockContextProvider(mockTimeProvider)
+        mockContextProvider.setup()
+        val mockCalendarProvider = MockCalendarProvider(mockContextProvider, mockTimeProvider)
+        mockCalendarProvider.setup()
+        
+        // Setup mock components
+        mockComponents = MockApplicationComponents(
+            contextProvider = mockContextProvider,
+            timeProvider = mockTimeProvider,
+            calendarProvider = mockCalendarProvider
+        )
+        mockComponents.setup()
+
+        mockContext = mockContextProvider.fakeContext
+    }
+    
+    @Test
+    @Ignore("figure out how to pass the mock context to Events corectly to call the mock that skips the file id. can proabably have a mock db like in this class but not needed for now")
+    fun testOriginalDismissEventWithValidEvent() {
+        // Given
+        val event = createTestEvent()
+        every { EventsStorage(mockContext).getEvent(event.eventId, event.instanceStartTime) } returns event
+        every { EventsStorage(mockContext).deleteEvent(event.eventId, event.instanceStartTime) } returns true
+        
+        // When
+        ApplicationController.dismissEvent(
+            mockContext,
+            EventDismissType.ManuallyDismissedFromActivity,
+            event.eventId,
+            event.instanceStartTime,
+            0,
+            false
+        )
+        
+        // Then
+        verify { EventsStorage(mockContext).getEvent(event.eventId, event.instanceStartTime) }
+        verify { EventsStorage(mockContext).deleteEvent(event.eventId, event.instanceStartTime) }
+    }
+    
+    @Test
+    @Ignore("figure out how to pass the mock context to Events corectly to call the mock that skips the file id. can proabably have a mock db like in this class but not needed for now")
+    fun testOriginalDismissEventWithNonExistentEvent() {
```

Comment:
We can probably add a mock based off of the one in this test to the fixtures to advance storage and the other storages and these should start working